lib/browser.js

```diff
@@ -101,6 +101,8 @@ class Browser {
     this.lastResult = new Result()
     this.lastResult.total = info.total
 
+    this.state = EXECUTING
+
```

Comment:
Changing `this.state` in client callbacks responding to events makes sense: we should to this uniformly unless there is some oddity which prevents us.  Therefore please try to remove the `this.state = EXECUTING` in the client command `execute()`.  The `execute()` command should not set the state; it should be set as you do here, in the start callback.

Reply:
I agree that `execute()` command shouldn't set state to `EXECUTING`, as actually tests are not executing at that moment and before `onStart()`. But we should make a difference between `READY` state of the browser and the state which is after `execute()` and before `onStart()`. We want at least the browser's methods `onKarmaError()` and `onInfo()`during this state. 

I've added additional state called CONFIGURING, which explains the actual state between `execute()` and `onStart()`. Is that OK?

Reply:
Yes! This is better. How about one more improvement?  The READY state is IMO misnamed.  After this change the states are:
```js
Browser.STATE_READY = READY
Browser.STATE_CONFIGURING = CONFIGURING
Browser.STATE_EXECUTING = EXECUTING
Browser.STATE_READY_DISCONNECTED = READY_DISCONNECTED
Browser.STATE_EXECUTING_DISCONNECTED = EXECUTING_DISCONNECTED
Browser.STATE_DISCONNECTED = DISCONNECTED
```
But I think the code would be clearer if the initial state was `CONNECTED` rather than `READY`:
```js
Browser.STATE_CONNECTED = CONNECTED
Browser.STATE_CONFIGURING = CONFIGURING
Browser.STATE_EXECUTING = EXECUTING
Browser.STATE_READY_DISCONNECTED = READY_DISCONNECTED
Browser.STATE_EXECUTING_DISCONNECTED = EXECUTING_DISCONNECTED
Browser.STATE_DISCONNECTED = DISCONNECTED
```
That clarifies the relationship with others states like `EXECUTING_DISCONNECTED` and avoids ambiguity about what the browser is "ready-for".

This version of the new state would be documented eg:
```js
// The browser is connected but not yet been commanded to execute tests.
  const CONNECTED = 1
```

Reply:
I don't have any strong opinion here about naming.
State `READY_DISCONNECTED` is actually never used (and sounds confusing with the new states). I'll remove it.